### PR TITLE
convert from Docker to Composite Github Action

### PR DESCRIPTION
**Pull request by davidbloss**
[Convert report-deploy-github-action to composite action](https://github.com/OpsLevel/team-platform/issues/430)

The Docker based approach to this Github Action causes friction for some customers operating their own custom Github runners. Composite actions are much simpler. We decided against taking the Javascript action route.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -8,35 +8,63 @@ inputs:
...
+    description: 'The service alias for the event'
     required: false
-    default: ''
+    default: ${{ github.repository }}
 runs:
-  using: 'docker'
-  image: 'Dockerfile'
+  using: "composite"
+  steps:
+    - name: Report Deploy to OpsLevel
```
Comment (rocktavious) [inline action.yml:42]: Is there a reason you don't put a step right before this that installs the CLI?

Comment (rocktavious) [reply action.yml:42]: Heres a first pass chatGPT took at it.  It looks about 85-90% correct but we probably need to tweak it after testing.  We should probably put this in the CLI repo and then you don't have to use homebrew and then you can do the easy step here of
```
curl -sLS https://raw.githubusercontent.com/OpsLevel/cli/main/install.sh | sudo sh
```

Here is the script
```
#!/bin/bash

# Version of the OpsLevel CLI to install
VERSION=${1:-v2024.07.1} # Default version, override by passing as an argument

# Function to determine the OS
detect_os() {
    case "$(uname -s)" in
        Linux*)     OS=linux;;
        Darwin*)    OS=macos;;
        CYGWIN*|MINGW*|MSYS*|MINGW32*|MINGW64*) OS=windows;;
        *)          echo "Unsupported OS"; exit 1;;
    esac
    echo "Detected OS: $OS"
}

# Function to determine the architecture
detect_arch() {
    case "$(uname -m)" in
        x86_64)    ARCH=amd64;;
        armv7l)    ARCH=armv7;;
        aarch64)   ARCH=arm64;;
        *)         echo "Unsupported architecture"; exit 1;;
    esac
    echo "Detected Architecture: $ARCH"
}

# Function to download and install the CLI tool
install_cli() {
    DOWNLOAD_URL="https://github.com/OpsLevel/cli/releases/download/${VERSION}/opslevel_${OS}_${ARCH}.tar.gz"
    
    # Temporary directory to store the download
    TMP_DIR=$(mktemp -d)

    echo "Downloading OpsLevel CLI from $DOWNLOAD_URL ..."
    curl -L -o "$TMP_DIR/opslevel.tar.gz" "$DOWNLOAD_URL"
    
    if [ $? -ne 0 ]; then
        echo "Download failed. Please check the version and try again."
        exit 1
    fi

    echo "Extracting the OpsLevel CLI..."
    tar -xzf "$TMP_DIR/opslevel.tar.gz" -C "$TMP_DIR"

    echo "Installing the OpsLevel CLI to /usr/local/bin ..."
    sudo mv "$TMP_DIR/opslevel" /usr/local/bin/

    if [ $? -ne 0 ]; then
        echo "Installation failed."
        exit 1
    fi

    echo "Cleaning up..."
    rm -rf "$TMP_DIR"

    echo "OpsLevel CLI installed successfully!"
}

# Main script execution
detect_os
detect_arch
install_cli

```

Comment (taimoorgit) [reply action.yml:42]: Good point, I forgot to consider that the CLI adds commit information in the report deploy request 🤔

File: action.yml [hunk 2]
```diff
@@ -8,35 +8,50 @@ inputs:
...
-    default: ''
+    default: 'Event Created by OpsLevel Github Action'
   environment:
     description: 'The environment for the event'
     required: false
-    default: ''
+    default: production
   number:
-    description: 'The deploy number for the event - Default: ${GITHUB_RUN_NUMBER}'
+    description: 'The deploy number for the event'
```
Comment (rocktavious) [inline action.yml:32]: do we not have access to `GITHUB_RUN_NUMBER` now?

**Review by rocktavious — APPROVED** [review]
NVM.  I see what you did - you elevated the defaults into the action.yml.  👍

**Pull request merged**